common/flatpak-dir.c

```diff
@@ -3463,6 +3526,7 @@ flatpak_dir_get_deploy_data (FlatpakDir        *self,
 
```

Comment:
I think we should call ensure_repo here. It should be safe (i.e. succeed normally) because we found the deploy dir, so there is an installation dir.